### PR TITLE
Add Range Maximum Query

### PR DESCRIPTION
**Pull request by itsamirhn**
### Describe your change:

I create new `RangeMinimumQuery` class for RMQ problem using Spare Table algorithm.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Erfaniaa** [comment]
Hi @Kush1101, can you please take a look at this?

File: dynamic_programming/range_maximum_query.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class RangeMaximumQuery:
+    def __init__(self, array: list) -> None:
+        """
+        Initialize RangeMaximumQuery with given array.
+
+        Parameters:
+            array: list[int]
+
+        Example:
+            >>> rmq = RangeMaximumQuery([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
```
Comment (poyea) [inline dynamic_programming/range_maximum_query.py:22]: One less indent?

File: dynamic_programming/range_maximum_query.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            >>> array = [1, 3, 5, 7, 9, 11]
+            >>> queries = [(0, 1), (1, 4), (0, 5), (2, 3), (0, 6)]
+            >>> rmq = RangeMaximumQuery(array)
+            >>> for l, r in queries:
+            ...     print(rmq.query(l, r))
+            1
+            7
+            9
+            5
+            11
```
Comment (poyea) [inline dynamic_programming/range_maximum_query.py:88]: One less indent?

File: dynamic_programming/range_maximum_query.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
+"""
+Range Maximum Query Problem:
+
+    Given an array of integers and q queries, for each query, find the maximum value in the range [l, r).
+    The array is 0-indexed and the queries are 0-indexed.
+
+    More details:
+        https://www.geeksforgeeks.org/range-maximum-query-using-sparse-table/
+
```
Comment (poyea) [inline dynamic_programming/range_maximum_query.py:9]:
```suggestion
```

Comment (Erfaniaa) [reply dynamic_programming/range_maximum_query.py:9]: I think "More details: https://www.geeksforgeeks.org/range-maximum-query-using-sparse-table/" is not necessary to be here.

**Comment by itsamirhn** [comment]
@poyea I fixed your suggestions, but now I don't realize why pre-commit tests are failing?!

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**